frontend/src/context/settings-up-to-date-context.tsx

```diff
@@ -0,0 +1,42 @@
+import React from "react";
```

Comment:
this seems really heavy for checking that settings are up-to-date. Is there a better way?

Reply:
This "context" needs to get loaded for the entire application every time, so maybe we just put this logic into the base component?

Reply:
This was my last option but the problem is we are fetching data from local storage. That means the functions to check whether it is up to date will only re-render if the component does, and in this case our sidebar does not re-render after a settings save mutation. For that reason I had to implement a context to properly manage the state

I think moving the logic into the base component would then require us to drill the set state action through a few prop/hooks, so better move it into a shared context to keep things simple

Reply:
OK--I think we can delete all this once everyone has transitioned